src/test/coins_tests.cpp

```diff
@@ -180,10 +183,10 @@ class CCoinsViewTest : public CCoinsView
         BatchWriteNullifiers(mapSproutNullifiers, mapSproutNullifiers_);
         BatchWriteNullifiers(mapSaplingNullifiers, mapSaplingNullifiers_);
 
-        mapCoins.clear();
         mapSproutAnchors.clear();
         mapSaplingAnchors.clear();
-        hashBestBlock_ = hashBlock;
+        if (!hashBlock.IsNull())
+            hashBestBlock_ = hashBlock;
         hashBestSproutAnchor_ = hashSproutAnchor;
         hashBestSaplingAnchor_ = hashSaplingAnchor;
```

Comment:
Similar to above, does it make sense to wrap these in if blocks?

Reply:
Looks good, although the lines
```
mapSproutAnchors.clear();
mapSaplingAnchors.clear();
```
have disappeared. I don't remember this the first time around. Has something changed?